### ISSUE TITLE
Truck Logo Image sizes

### ISSUE THREAD

**Issue by Lowell20** [codeforeauclaire/foodtrucks#41]
Let's decide on what size and aspect ratio we want for the logo images.
I can provide resized image paths in the json automatically which might provide a more consistent experience for the consumer javascript.

We should also decide in some kind of image upload parameters for the food truck users in the admin ui.

**Comment by AnthonyAstige** [comment]
Sounds good to address the aspect ratio for a consistent consumer ux.

I expect challenges in getting the truck vendors to adhere to a standard, and just resizing or auto-cropping often yeilds sub-par results. Is there an easy Drupal drop in for cropping images in the browser? Alternatively since there won't be that many trucks, and I don't expect them to update their logo too often, we could just allow them to upload whatever and ping one of us to 'clean it up' afterwards.

**Comment by Lowell20** [comment]
right, exactly. So I am asking to know what that size and aspect ratio should be. 
We can enforce some restriction on the logo upload form
There looks like a working image crop for Drupal 8, haven't tested in yet though
And it is easy to scale and crop automatically, although you are right about sub-par results many times

**Comment by AnthonyAstige** [comment]
@aaronsalmon It appears the logo aspect ratios in [your Balsamic](https://cloud.githubusercontent.com/assets/236039/15490350/bfccfd9a-212c-11e6-94bc-f4c3122868f5.png) differ between the Vendor and Consumer sides. The consumer side is our target for which the balsamic has a wide image, so perhaps we should lock in on a simple and [common aspect ratio](https://photo.stackexchange.com/questions/33713/most-common-aspect-ratio) like 3:2?

@Lowell20 As far as sizing what we keep on the drupal servers I lean towards keeping a bit larger than we need, perhaps 400 pixels wide with reasonable compression; just let the browser downsize the dimensions matching the same ratio.  Keeps things simple so we don't prematurely optimize.

**Comment by AnthonyAstige** [comment]
@Lowell20 

> We can enforce some restriction on the logo upload form

If you mean [data size upload restriction](https://stackoverflow.com/questions/9153224/how-to-limit-file-upload-type-file-size-in-php) I think it's preferable to just let the user here upload whatever and compress ourselves (be it manually or automatically, including re-sizing dimensions or not).  If you mean some other Drupal magic then great we'll have to see what it is.
